internal/transport/transport_test.go

```diff
@@ -474,12 +478,48 @@ func setUpWithNoPingServer(t *testing.T, copts ConnectOptions, connCh chan net.C
 			close(connCh)
 			return
 		}
+		if _, err := io.ReadFull(conn, make([]byte, len(clientPreface))); err != nil {
+			t.Errorf("Error while reading client preface: %v", err)
+			return
+		}
 		framer := http2.NewFramer(conn, conn)
 		if err := framer.WriteSettings(); err != nil {
-			t.Errorf("Error at server-side while writing settings: %v", err)
+			t.Errorf("Error at server-side while writing settings: %v", framer.ErrorDetail())
```

Comment:
is `framer.ErrorDetail()` inclusive of `err` ?

Reply:
Yes, it is and it's more detailed